linera-faucet/server/src/tests.rs

```diff
@@ -272,3 +277,415 @@ async fn test_batch_size_reduction_on_limit_errors() {
+        request_notifier: Arc::clone(&request_notifier_2),
+        storage: client.storage_client().clone(),
+    };
+
+    // Create new batch processor for the second instance
+    let batch_config_2 = super::BatchProcessorConfig {
+        amount: Amount::from_tokens(1),
+        end_timestamp: Timestamp::from(6000),
+        start_timestamp: Timestamp::from(0),
+        start_balance: Amount::from_tokens(6),
+        max_batch_size: 1,
+    };
+    let batch_processor_2 = super::BatchProcessor::new(
+        batch_config_2,
+        Arc::clone(&context),
+        client.clone(),
+        Arc::clone(&faucet_storage_2),
+        Arc::clone(&pending_requests_2),
+        Arc::clone(&request_notifier_2),
+    );
+
+    // Start the new batch processor
+    let cancellation_token_2 = CancellationToken::new();
+    let processor_task_2 = {
+        let mut batch_processor = batch_processor_2;
+        let token = cancellation_token_2.clone();
+        tokio::spawn(async move { batch_processor.run(token).await })
+    };
+
+    // === PHASE 4: Verify blockchain sync restored the correct mappings ===
+
+    // Test that the blockchain sync correctly restored the chain mappings
+    let chain_1_after_sync = root_2
+        .do_claim(test_owner_1)
+        .await
+        .expect("Should return existing chain after blockchain sync");
+    assert_eq!(
+        chain_1_id,
+        chain_1_after_sync.id(),
+        "Chain ID should be correctly restored from blockchain for owner 1"
+    );
+
+    let chain_2_after_sync = root_2
+        .do_claim(test_owner_2)
+        .await
+        .expect("Should return existing chain after blockchain sync");
+    assert_eq!(
+        chain_2_id,
+        chain_2_after_sync.id(),
+        "Chain ID should be correctly restored from blockchain for owner 2"
+    );
+
+    // === PHASE 5: Verify new claims still work correctly ===
+
+    // Verify that a completely new owner can still claim a new chain
+    clock.set(Timestamp::from(3000));
+    let test_owner_3 = AccountPublicKey::test_key(102).into();
+    let chain_3 = root_2
+        .do_claim(test_owner_3)
+        .await
+        .expect("New owner should be able to claim after sync");
+
+    // Verify the new chain is different from the existing ones
+    assert_ne!(
+        chain_3.id(),
+        chain_1_id,
+        "New chain should have different ID from synced chains"
+    );
+    assert_ne!(
+        chain_3.id(),
+        chain_2_id,
+        "New chain should have different ID from synced chains"
+    );
+
+    // Verify that the new chain mapping is also persisted
+    let chain_3_again = root_2
+        .do_claim(test_owner_3)
+        .await
+        .expect("Re-claim should return the new chain");
+    assert_eq!(
+        chain_3.id(),
+        chain_3_again.id(),
+        "New chain should be persistent after sync"
+    );
+
+    // Clean up
+    cancellation_token_2.cancel();
+    let _ = processor_task_2.await;
+}
```

Comment:
nice! what was the prompt that you used to ask CC to write these tests?

Reply:
It was very high-level. Something like: "Write a test that claims a chain, then restarts the faucet, and claims a chain with the same account again, to verify that even after a restart the mapping is preserved."